mixer/cmd/root.go

```diff
@@ -0,0 +1,97 @@
+// Copyright © 2017 NAME HERE <EMAIL ADDRESS>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/clearlinux/mixer-tools/builder"
+
+	"github.com/spf13/cobra"
+)
+
+var config string
+
+// RootCmd represents the base command when called without any subcommands
+var RootCmd = &cobra.Command{
+	Use:  "mixer",
+	Long: `Mixer is a tool used to compose OS update content and images.`,
+}
+
+type initCmdFlags struct {
+	all         bool
+	clearver    int
+	mixver      int
+	upstreamurl string
+}
+
+var initFlags initCmdFlags
+
+var initCmd = &cobra.Command{
+	Use:   "init-mix",
+	Short: "Initialize the mixer and workspace",
+	Long:  `Initialize the mixer and workspace`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		b := builder.New()
+		b.LoadBuilderConf(config)
+		b.ReadBuilderConf()
+		return b.InitMix(strconv.Itoa(initFlags.clearver), strconv.Itoa(initFlags.mixver), initFlags.all, initFlags.upstreamurl)
+	},
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
```

Comment:
I'd just inline this into our main function.

Reply:
Unless there is a specific reason, the content in "root.go" should just be part of main.go IMHO.

Reply:
This is simply the intended structure cobra generates when starting a new project.